### PR TITLE
feat: #3 validate commit msgs and automate semver bumps, CHANGELOG updates, and npm publication

### PR DESCRIPTION
**Pull request by gregswindle**
### 1.1. :heavy_check_mark: [`commitplease`](https://github.com/jzaefferer/commitplease)

1.1.1. [`package.json` is configured](https://www.npmjs.com/package/commitplease#angularjs) for AnguarJS conventions

1.1.2. Enforces the [AngularJS Git Commit Guidelines](https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit)

### 1.2. ✔️ [`semantic-release`](https://github.com/semantic-release/semantic-release)

`semantic-release` is a fully automated package publishing library.  When used with `commitplease`, we should be able to communicate consistently and meaningfully, as well as automate difficult tasks like CHANGELOG.md updates.

### :heavy_minus_sign: [`commitizen`](https://www.npmjs.com/package/commitizen)

`commitizen` looks very promising, but it requires global installation. We should recommend it, regardless.

> #### ℹ️ `commitizen` can (and should) be used alongside `commitplease` and `semantic-release`.
>
> It's not part of this PR, since `commitizen` must be installed globally.

## 3. Deferred acceptance criteria

I recommend we defer automatic installation and setup of

### :hourglass: 3.2. [`semantic-release`](https://github.com/semantic-release/semantic-release)

In other words, when engineers run

```bash
$ yo angular-library
```

`commitplease` and `semantic-release` will not be part of installation. We need to refactor for PR #4 first, in my opinion.

> ###  :imp: PR #4 should be completed first
> To enable criteria (3.1.) and (3.2.), we'll have to enable [composability](http://yeoman.io/authoring/composability.html).
>
> It might make more sense to update all dependencies before this, since the `generator-android-library` currently depends on a very old version of `yeoman-generator`: `v0.22.6`.

### CODE REVIEW

**Comment by gregswindle** [comment]
@mkorat1717 I just tried to scaffold a project, and it didn't work. :flushed: Is it working for you?

**Comment by mkorat1717** [comment]
Just tested- looks like it didn't work.

**Comment by gregswindle** [comment]
@mkorat1717 can you please help me debug this PR's branch?

I add an new wiki page called [_How to debug Yeoman generators_](https://github.com/gregswindle/generator-android-library/wiki/How-to-debug-Yeoman-generators)  to get you started.
